packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDomainMapAndImporter.cpp

```diff
@@ -146,59 +181,135 @@ namespace {
-    if (comm->getSize() > 1) {
-      MT norm = STM::zero ();
-
-      // we know the map is contiguous...
-      const size_t NumMyElements = (comm->getRank () == 0) ?
-        A.getDomainMap ()->getGlobalNumElements () : 0;
-      RCP<const Map<LO,GO,Node> > NewMap =
-        rcp (new Map<LO,GO,Node> (INVALID, NumMyElements, ZERO, comm));
-      RCP<const Tpetra::Import<LO,GO,Node> > NewImport =
-        rcp (new Import<LO,GO,Node> (NewMap, A.getColMap ()));
-
-      B.replaceDomainMapAndImporter (NewMap, NewImport);
-
-      // Fill a random vector on the original map
-      Vector<Scalar,LO,GO,Node> AVecX(A.getDomainMap());
-      AVecX.randomize();
-
-      // Import this vector to the new domainmap
-      Vector<Scalar,LO,GO,Node> BVecX(B.getDomainMap());
-      Tpetra::Import<LO,GO,Node> TempImport(A.getDomainMap(),NewMap); // (source,target)
-      BVecX.doImport(AVecX,TempImport,Tpetra::ADD);
-
-      // Now do some multiplies
-      Vector<Scalar,LO,GO,Node> AVecY(A.getRangeMap());
-      Vector<Scalar,LO,GO,Node> BVecY(B.getRangeMap());
-      A.apply(AVecX,AVecY);
-      B.apply(BVecX,BVecY);
-
-      BVecY.update (-STS::one (), AVecY, STS::one ());
-      norm = BVecY.norm2();
-
-      out << "Residual 2-norm: " << norm << endl
-          << "Residual 1-norm: " << BVecY.norm1 () << endl
-          << "Residual Inf-norm: " << BVecY.normInf () << endl;
-
-      // Macros don't like spaces, so we put the test outside the
-      // macro.  Use <= rather than <, so the test passes even if
-      // Scalar is an integer type.
-      //
-      // FIXME (mfh 10 Mar 2013) We should pick the tolerance relative
-      // to the Scalar type and problem size.
-      const bool normSmallEnough = norm <= as<MT> (1e-10);
-      TEST_EQUALITY ( normSmallEnough, true );
+    // we know the map is contiguous...
+    const size_t NumMyElements = (comm->getRank () == 0) ?
+      A.getDomainMap ()->getGlobalNumElements () : 0;
+    RCP<const Map<LO,GO,Node> > NewMap =
+      rcp (new Map<LO,GO,Node> (INVALID, NumMyElements, ZERO, comm));
+
+    B.replaceDomainMap (NewMap);
+
+    applyAndCheckResult(A, B, NewMap);
+  }
+ 
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( CrsMatrix, ReplaceDomainMapAndImporter, LO, GO, Scalar, Node )
+  {
+    // Based on the FullTriDiag tests...
+
+    typedef CrsMatrix<Scalar,LO,GO,Node> MAT;
+    typedef ScalarTraits<Scalar> STS;
+
+    const size_t ONE  = OrdinalTraits<size_t>::one();
+    const size_t ZERO = OrdinalTraits<GO>::zero();
+    const global_size_t INVALID = OrdinalTraits<global_size_t>::invalid();
+    // get a comm
+    RCP<const Comm<int> > comm = Tpetra::getDefaultComm();
+    const size_t numImages = comm->getSize();
+    const size_t myImageID = comm->getRank();
+    if (numImages < 3) return;
+    // create a Map
+    RCP<const Map<LO,GO,Node> > map = createContigMapWithNode<LO,GO,Node>(INVALID,ONE,comm);
+
+    // RCP<FancyOStream> fos = Teuchos::fancyOStream(rcp(&std::cout,false));
```

Comment:
seems like a leftover line that can be deleted?

Reply:
Done